File: app/src/main/java/com/studyasist/data/local/entity/Attempt.kt
package com.studyasist.data.local.entity

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(
    tableName = "attempts",
    foreignKeys = [
        ForeignKey(
            entity = Assessment::class,
            parentColumns = ["id"],
            childColumns = ["assessmentId"],
            onDelete = ForeignKey.CASCADE
        )
    ],
    indices = [Index(value = ["assessmentId"])]
)
data class Attempt(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0,
    val assessmentId: Long,
    val startedAt: Long,
    val endedAt: Long? = null,
    val userNotes: String? = null,
    val needsManualReview: Boolean = false
)

File: StudyAsist-iOS/StudyAsist/Models/Attempt.swift
//
//  Attempt.swift
//  StudyAsist
//

import Foundation
import SwiftData

@Model
final class Attempt {
    @Attribute(.unique) var id: Int64
    var assessmentId: Int64
    var startedAt: Int64
    var endedAt: Int64?
    var userNotes: String?
    var needsManualReview: Bool
    
    init(
        id: Int64 = 0,
        assessmentId: Int64,
        startedAt: Int64,
        endedAt: Int64? = nil,
        userNotes: String? = nil,
        needsManualReview: Bool = false
    ) {
        self.id = id
        self.assessmentId = assessmentId
        self.startedAt = startedAt
        self.endedAt = endedAt
        self.userNotes = userNotes
        self.needsManualReview = needsManualReview
    }
}
